packages/gatsby/src/bootstrap/load-plugins/load.js

```diff
@@ -130,7 +130,7 @@ module.exports = async (config = {}) => {
   // Add internal plugins
   const internalPlugins = [
     `../../internal-plugins/dev-404-page`,
-    `../../internal-plugins/component-page-creator`,
+    `../../internal-plugins/component-page-creator`, // ???
```

Comment:
I believe we should remove this however I do not know how to require the new plugin and supply and options. My understanding is that this would be supplied by user in the `gatsby-config.js` file.

Reply:
We'd keep `src/pages` as where components convert to pages automatically. That's one built-in constant we have.

What breaking this out to a plugin accomplishes is it's now easy to setup the same thing for other pages.